### PR TITLE
Added Optimizely tags and custom event tracking

### LINKED ISSUE #1186: Add Optimizely tags and conversion tracking to repo
Before starting a A/B testing we'll have to set up Optimizely first,

> ### Setup Optimizely

> Step one is to get the Optimizely tag into the head of the template, like this https://github.com/mozilla/donate.mozilla.org/blob/master/pages/index.jsx#L8

> Then setup Optimizely conversion tracking on the goal you are optimizing for. [Donate example here](https://github.com/mozilla/donate.mozilla.org/blob/master/public/js/analytics.js#L121) and [Optimizely Docs](https://help.optimizely.com/hc/en-us/articles/200039925-Custom-event-goals). I suspect this is `number of pledges to teach`

for more details see [Adam's comment](https://github.com/mozilla/teach.webmaker.org/issues/1172#issuecomment-132505377) on https://github.com/mozilla/teach.webmaker.org/issues/1172

### PR DESCRIPTION
**Pull request by mmmavis**
This fixes #1186.

@adamlofting 
I wasn't sure if we need to add `optimizelysubdomain.jsx` like [what you did for donate.mozilla.org](https://github.com/mozilla/donate.mozilla.org/pull/325/files)... so I left it out. :grin: Could you briefly describe what that file(or cookie) is for? I couldn't find relevant info on Optimizely docs. :bowtie:

### CODE REVIEW

File: pages/home.jsx [hunk 1]
```diff
@@ -277,6 +277,9 @@ var HomePage = React.createClass({
   componentDidMount: function() {
     if (this.context.router.getCurrentQuery().pledge === "thanks") {
```
Comment (mmmavis) [inline pages/home.jsx]: this is where the BSD thank you page gets redirected to

File: components/optimizely.jsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+var React = require('react');
+
+var optimizelyActive = process.env.OPTIMIZELY_ACTIVE || 'yes';
+var optimizelyID = process.env.OPTIMIZELY_ID || '206878104';
+var optimizelyURL = 'https://cdn.optimizely.com/js/' + optimizelyID + '.js';
+
+var Optimizely = React.createClass({
+  render: function() {
```
Comment (Pomax) [inline components/optimizely.jsx]: `return optimizelyActive ?  <script src={optimizelyURL} /> : false;`?

**Comment by adamlofting** [comment]
Hey Mavis,

> I wasn't sure if we need to add optimizelysubdomain.jsx like what you did for donate.mozilla.org... so I left it out. :grin: Could you briefly describe what that file(or cookie) is for? I couldn't find relevant info on Optimizely docs. :bowtie:

Sorry, I didn't get time to explain that one properly in my previous tickets, but yet, we do need to include it as an extra security measure because we're sharing the mozilla.org domain with many sites. There's some further info here: https://bugzilla.mozilla.org/show_bug.cgi?id=1125995

File: components/optimizely.jsx (hunk 2 above)
Comment (adamlofting) [inline components/optimizely.jsx]: :+1: that's nicer than my version :smile:

**Comment by adamlofting** [comment]
This looks great, but let's add in the subdomain aspect, and Pomax's suggestion for the Optimizely.jsx component. Then I'm happy to quickly R? again.

**Comment by mmmavis** [comment]
Thanks for the feedback @Pomax and @adamlofting . I updated the PR again.
@adamlofting R? :smile_cat:

File: components/optimizely-subdomain.jsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+var React = require('react');
+
+var optimizelyActive = process.env.OPTIMIZELY_ACTIVE || 'yes';
+var cookieDomain = process.env.FULL_SUBDOMAIN_FOR_COOKIE || 'donate.mozilla.org';
+
+var OptimizelySubdomain = React.createClass({
+  cookieScript: function() { return {__html: "window['optimizely'] = window['optimizely'] || []; window['optimizely'].push(['setCookieDomain', '" + cookieDomain + "']);"}; },
+  render: function() {
+    return optimizelyActive ? <script dangerouslySetInnerHTML={this.cookieScript()} /> : false;
```
Comment (Pomax) [inline components/optimizely-subdomain.jsx]: hmm, would this work? (not 100% sure):
```
var React = require('react');

var optimizelyActive = process.env.OPTIMIZELY_ACTIVE || 'yes';
var cookieDomain = process.env.FULL_SUBDOMAIN_FOR_COOKIE || 'donate.mozilla.org';

var OptimizelySubdomain = React.createClass({
  render: function() {
    return optimizelyActive ? <script>
      window['optimizely'] = window['optimizely'] || [];
      window['optimizely'].push(['setCookieDomain', '{cookieDomain}']);"};
    </script> : false;
  }
});

module.exports = OptimizelySubdomain;

```

File: components/optimizely-subdomain.jsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+var React = require('react');
+
+var optimizelyActive = process.env.OPTIMIZELY_ACTIVE || 'yes';
```
Comment (Pomax) [inline components/optimizely-subdomain.jsx]: `!!process.env.OPTIMIZELY_ACTIVE || true`?

File: components/optimizely-subdomain.jsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+var cookieDomain = process.env.FULL_SUBDOMAIN_FOR_COOKIE || 'donate.mozilla.org';
```
Comment (mmmavis) [inline components/optimizely-subdomain.jsx]: oops, domain should be `teach.mozilla.org` instead.  will fix.

File: components/optimizely-subdomain.jsx (hunk 3 above)
Comment (mmmavis) [inline components/optimizely-subdomain.jsx]: @Pomax I tried but it seems like special characters were escaped -- source code shows 
```html
<script>
window[&#x27;optimizely&#x27;] = window[&#x27;optimizely&#x27;] || []; 
window[&#x27;optimizely&#x27;].push([&#x27;setCookieDomain&#x27;, &#x27;teach.mozilla.org&#x27;]);
</script>
```

Comment (Pomax) [inline components/optimizely-subdomain.jsx]: ah. In that case let's just indent the `cookieScript` a little and keep the `render` as is

**Comment by Pomax** [comment]
I don't see anything blowing up. R+

**Comment by mmmavis** [comment]
yay yay yay merginggggggggg

**Pull request merged**